jerry-core/ecma/builtin-objects/ecma-builtin-date-prototype.cpp

```diff
@@ -42,6 +43,53 @@
  */
 
 /**
+ * Insert leading zeros to a string of a number if needed.
+ */
+static void
+ecma_date_insert_leading_zeros (ecma_string_t **str_p, /**< input/output string */
+                                ecma_number_t num, /**< input number */
+                                uint32_t length) /**< length of string of number */
+{
+  JERRY_ASSERT (length >= 1);
+
+  /* If the length is bigger than the number of digits in num, then insert leding zeros. */
+  for (uint32_t i = length - 1; i > 0 && num < pow (10,i); i--)
```

Comment:
Why do we need to perform `pow` on every iteration? It is unnecessarily resource consuming.